### ISSUE TITLE
Continue analyze process on zip file error

### ISSUE THREAD

**Issue by guyacosta** [microsoft/ApplicationInspector#196]
Presently, an error in decompressing a particular file embedded in an archive type causes the processing to halt and report the error.  A suggestion was made to continue processing other files in the archive while reporting the error.  

One thought might be to have a max or fatal error limit.

The implementation would need to add failed file metadata to the FileEntry object but mark it as "bad" which would be checked etc.  While not necessarily a bug in the decompression i.e. file may be corrupt, this could be an enhancement if enough people are needing this.

**Comment by gfs** [comment]
I suggest having an output field for failing analysis of a single file or archive (i.e. not actually a fatal error overall) and to continue execution unless you absolutely can't. Otherwise scanning large quantities of files which may contain failures has many interruptions.

-------- Original Message --------

**Comment by Markkuu** [comment]
+1 for this. Currently having this issue with scanning. Would be preferable to have a flag to continue unless absolutely necessary to crash out.

**Comment by gfs** [comment]
@markkuu I have some improvements for this in #204 and we will have more on that once microsoft/OSSGadget#47 is complete.

**Comment by gfs** [comment]
@guyacosta Per your initial desecription
> The implementation would need to add failed file metadata to the FileEntry object but mark it as "bad" which would be checked etc. While not necessarily a bug in the decompression i.e. file may be corrupt, this could be an enhancement if enough people are needing this.

I don't think this is needed. If multiextractor doesn't know how to extract the file properly its not useful to Application Inspector.  If you want a "export invalid entries" option I'd add a new issue to OSSGadget.  I think we can call this fixed (durable to malformed archives) with microsoft/OSSGadget#47 and once we update to use the published nupkg of multiextractor.

**Comment by gfs** [comment]
This is complete. I brought over the engine from upstream multi extractor.
